### PR TITLE
chore: Checksum releases

### LINKED ISSUE #1553: PGP-sign your releases
I hope you understand that publishing hashes on Github offers very weak assurances of the authenticity of the software. please create a release signing PGP key controlled by your main developers and sign the files you release.

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Partly resolves #1553 

~Signs the github releases~, generates a SHA256 checksum and uploads the file with deployment.

### CODE REVIEW

File: .github/workflows/deploy-release.yml [hunk 1]
```diff
@@ -5,7 +5,69 @@ on:
...
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+          git_tag_gpgsign: true
+
+      - name: Download source
+        run: curl -sL "${{ github.event.release.tarball_url }}" >"$ARCHIVE_NAME".tar.gz
```
Comment (katspaugh) [inline .github/workflows/deploy-release.yml:26]: The idea was to sign & checksum builds, not the source code.

File: .github/workflows/deploy-release.yml [hunk 2]
```diff
@@ -5,7 +5,86 @@ on:
...
+
+      - name: Unzip source
+        run: mkdir "$ARCHIVE_NAME" && tar -xzf "$ARCHIVE_NAME".tar.gz -C "$ARCHIVE_NAME" --strip-components 1
+
+      # Remove everything except the build files
+      - name: Clean source
+        run: |
+          mkdir tempout
+          cd $ARCHIVE_NAME
+          mv ./out/* ../tempout
```
Comment (katspaugh) [inline .github/workflows/deploy-release.yml:36]: I highly doubt an archive with the source code contains the build directory...

File: .github/workflows/deploy-release.yml [hunk 3]
```diff
@@ -23,17 +25,75 @@ jobs:
...
+        run: tar -czf "$ARCHIVE_NAME".tar.gz out
+
+      - name: Create checksum
+        run: sha256sum "$ARCHIVE_NAME".tar.gz > ${{ env.ARCHIVE_NAME }}-sha256-checksum.txt
+
+      # Upload the checksum file so that it can be reused in the next job
+      - uses: actions/upload-artifact@v2
+        with:
+          name: checksum-file
+          path: ${{ env.ARCHIVE_NAME }}-sha256-checksum.txt
```
Comment (katspaugh) [inline .github/workflows/deploy-release.yml:46]: Can't it be uploaded straight as a release asset? I don't see it used for anything else.

**Review by katspaugh — APPROVED** [review]
👍 
Looks good, just waiting for @luarx to add a GPG key now.

**Comment by usame-algan** [comment]
As discussed, I removed the signing related part until we can make a decision. This PR will only checksum the build and upload it with deployment so there is an integrity check.

**Pull request merged**